src/bam/collapse_reads_to_fragments/calc_consensus.rs

```diff
@@ -49,17 +49,37 @@ impl<'a> CalcOverlappingConsensus<'a> {
         let read_orientations_opt = self.build_read_orientation_string();
         let mut consensus_lh = LogProb::ln_one();
         for i in 0..seq_len {
-            let likelihoods = ALLELES
-                .iter()
-                .map(|a| Self::overall_allele_likelihood(self, a, i))
-                .collect_vec(); //This will be calculated every iteration
-            Self::build_consensus_sequence(
-                likelihoods,
-                &mut consensus_lh,
-                &mut consensus_seq,
-                &mut consensus_qual,
-                33.0,
-            );
+            match (
+                self.recs1().len() == 1,
+                self.map_read_pos(i, self.r1_vec()),
+                self.map_read_pos(i, self.r2_vec()),
+            ) {
+                (true, Some(base_pos), None) => {
+                    let base = self.recs1()[0].seq().as_bytes()[base_pos];
+                    consensus_seq.push(base);
+                    consensus_qual.push(self.recs1()[0].qual()[base_pos] + 33);
+                    consensus_lh += Self::overall_allele_likelihood(self, &base, i);
```

Comment:
The likelihood of the consensus reads is updated here for sake of completeness.
Still, the overall sequence likelihood is never required for consensus calling on bam files. Only for fastq files when estimating the best insert size.
Therefore it would not be necessary to update the likelihood here. 
Should this still be included for remaining a valid likelihood or should we drop calculating the sequence likelihood for bam files and implement this as an option in functions shared by bam- and fastq-consensus calculation.

Reply:
You can keep this. I don't think this will cause a lot of additional runtime.